### PR TITLE
Ksl hacks/move repositories

### PR DESCRIPTION
**Pull request by KSLHacks**
Opened new PR from KSLHacks/moveRepository into dev.
#74 was into master.

### CODE REVIEW

File: JabbR-Core.sln [hunk 1]
```diff
@@ -13,8 +13,11 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "JabbR-Core", "src\JabbR-Core\JabbR-Core.xproj", "{44F3E272-7DD2-4F54-A013-DEBADFA29CE8}"
 EndProject
+<<<<<<< HEAD
```
Comment (adamtuliper) [inline JabbR-Core.sln]: note markers

File: src/JabbR-Core/Commands/CommandManager.cs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            }
+
+            command.Execute(context, callerContext, args);
+
+            return true;
+        }
+
+        private void MatchCommand(string commandName, out ICommand command)
+        {
+            throw new NotImplementedException();
```
Comment (adamtuliper) [inline src/JabbR-Core/Commands/CommandManager.cs]: I msgd on over a couple weeks ago some code to try here, give it a whirl. Then we can get commented blocks below removed.

File: src/JabbR-Core/wwwroot/Chat.js [hunk 3]
```diff
@@ -999,7 +999,7 @@
     };
 
     chat.client.showUsersOwnedRoomList = function (user, rooms) {
-        if (rooms.length === 0) {
+        if (rooms.length === 0) {op
```
Comment (adamtuliper) [inline src/JabbR-Core/wwwroot/Chat.js]: whats the 'op' here from?

Comment (KSLHacks) [inline src/JabbR-Core/wwwroot/Chat.js]: Fixed.

File: JabbR-Core.sln (hunk 1 above)
Comment (KSLHacks) [inline JabbR-Core.sln]: Resolved .sln file

**Comment by KSLHacks** [comment]
Working with Justine to integrate the working implementation of data package into our branch.

File: src/JabbR-Core/Commands/CommandManager.cs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        //}
+
+        private static IList<ICommand> GetCommands()
+        {
+            // Use MEF to locate the content providers in this assembly
+            //var catalog = new AssemblyCatalog(typeof(CommandManager).Assembly);
+            //var compositionContainer = new CompositionContainer(catalog);
+            //return compositionContainer.GetExportedValues<ICommand>().ToList();
+            return new List<ICommand>() { new JoinCommand() };
+        }
```
Comment (KSLHacks) [inline src/JabbR-Core/Commands/CommandManager.cs]: @adamtuliper This was the workaround we used. We Manually populated JoinCommand() in the List<ICommand>.
Lines 143 and 199 use `using System.ComponentModel.Composition.Hosting;` which is not supported by .NET Core. 

Currently trying to find workaround using System.Reflection

**Comment by KSLHacks** [comment]
@adamtuliper Ready for a merge if you could take quick look. Justine and I decided not to integrate yet since we still do not know for sure if the data package models is what we will end up using. Any changes to the models will result in Hub.chat.cs changes. To avoid more work after every change, we think it's best to keep it separate and since this branch has a lot of changes we should merge and update dev.

@JamesEarle Assigning you to test:
+Jabbr-Core project should be independent of the data package (still using Jabbr-Core.models).
+Home page should load successfully.
+Upon 'openRoom' click, a new loading screen should open up.
+Will not get past the loading screen on this PR.
+Console Output will display upon clicking to enter the room:
```
[13:13:00 GMT-0500 (Central Daylight Time)] SignalR: Triggering client hub event 'joinRoom' on hub 'Chat'.
VM660:3 Uncaught ReferenceError: Count is not defined
```

**Comment by JamesEarle** [comment]
All tests check out, works exactly as you said @KSLHacks. 

:shipit: @adamtuliper

**Comment by adamtuliper** [comment]
ignore review notes, creating issue to address so we can get this merged.

**Pull request merged**